group_project_v2/stage_components.py

```diff
@@ -254,23 +282,35 @@ def submission_review_view(self, context):
         # element
         return fragment
 
+    def _validate_upload(self, request):
+        if not self.stage.available_now:
+            template = messages.STAGE_NOT_OPEN_TEMPLATE if not self.stage.is_open else messages.STAGE_CLOSED_TEMPLATE
+            # 422 = unprocessable entity
+            return 422, {'result': 'error', 'message': template.format(action=self.stage.STAGE_ACTION)}
+
+        if not self.stage.is_group_member and not self.stage.is_admin_grader:
+            # 403 - forbidden
+            return 403, {'result': 'error', 'message': messages.NON_GROUP_MEMBER_UPLOAD}
+
+        try:
+            self.validator(request.params[self.upload_id].file)
+        except ValidationError as validationError:
+            message = validationError.message % validationError.params
+            # 400 - BAD REQUEST
+            return 400, {'result': 'error', 'message': message}
+
+        return None, None
```

Comment:
@snowcrshd Makes sense, but definitely not in scope of this deliverable.

This is actually a quick and more or less clean refactoring to support different validation protocol provided by the `validator` - other checks are just booleans, but this one is an exception with meaningful message, so this was the cleanest solution I was able to come up with in 5 minutes (and I didn't want to spend more than that on this). Suggestions welcome.